ui/client/components/modals/CustomActionDialog.tsx

```diff
@@ -0,0 +1,78 @@
+import React, {useCallback, useState} from "react"
+import {useDispatch, useSelector} from "react-redux"
+import {getProcessId} from "../../reducers/selectors/graph"
+import {loadProcessState} from "../../actions/nk"
+import GenericModalDialog from "./GenericModalDialog"
+import Dialogs from "./Dialogs"
+import HttpService from "../../http/HttpService"
+import {getModalDialog} from "../../reducers/selectors/ui"
+import {editors} from "../graph/node-modal/editors/expression/Editor"
+import {ExpressionLang} from "../graph/node-modal/editors/expression/types"
+
+function CustomActionDialog(): JSX.Element {
+
+  const processId = useSelector(getProcessId)
+  const dispatch = useDispatch()
+  const action = useSelector(getModalDialog).customAction
+
+  const empty = (action?.parameters || []).reduce((obj, param) => ({...obj, [param.name]: ""}), {})
+
+  const [mapState, setState] = useState(empty)
+
+  const init = useCallback(() => setState(empty), [])
+
+  const confirm = useCallback(async () => {
+    await dispatch(HttpService
+      .customAction(processId, action.name, mapState)
+      .finally(() => dispatch(loadProcessState(processId))))
+  }, [processId, action, mapState])
+
+  const renderFieldLabel = (label) => {
```

Comment:
I think it gives nothing positive to have this function, its cleaner to have this div in place or extract it as component.

Reply:
:+1: true, it was copied from elswhere...